salt/states/dockerng.py

```diff
@@ -0,0 +1,1495 @@
+                    .format(image_path, desired_image, exc)
+                )
+                return ret
+            if image_update.get('Layers', []):
+                ret['changes']['load'] = image_update
+        comments.append('Image \'{0}\' was {1}'.format(desired_image, action))
+
+    if not pre_config:
+        pre_state = None
+    else:
+        pre_state = __salt__['docker-ng.state'](name)
+
+    if new_container:
+        if pre_config:
+            # Container exists, stop if necessary, then remove and recreate
+            if pre_state != 'stopped':
+                result = __salt__['docker-ng.stop'](name,
+                                                    timeout=stop_timeout,
+                                                    unpause=True)['result']
+                if result is not True:
+                    comments.append(
+                        'Container was slated to be replaced, but the '
+                        'container could not be stopped.'
+                    )
+                    ret['comment'] = _format_comments(comments)
+                    return ret
+
+            # Remove existing container
+            removed_ids = __salt__['docker-ng.rm'](name)
+            if not removed_ids:
+                comments.append('Failed to remove container {0}'.format(name))
+                ret['comment'] = _format_comments(comments)
+                return ret
+
+            # Removal was successful, add the list of removed IDs to the
+            # changes dict.
+            ret['changes'].setdefault('container', {})['removed'] = removed_ids
+
+        try:
+            # Create new container
+            create_result = __salt__['docker-ng.create'](
+                desired_image,
+                name=name,
+                client_timeout=client_timeout,
+                # Already validated input
+                validate_input=False,
+                **create_kwargs
+            )
+        except Exception as exc:
+            comments.append('Failed to create new container: {0}'.format(exc))
+            ret['comment'] = _format_comments(comments)
+            return ret
+
+        # Creation of new container was successful, add the return data to the
+        # changes dict.
+        ret['changes'].setdefault('container', {})['added'] = create_result
+
+    if new_container or pre_state != 'running':
+        try:
+            # Start container
+            __salt__['docker-ng.start'](
+                name,
+                # Already validated input earlier, no need to repeat it
+                validate_ip_addrs=False,
+                validate_input=False,
+                **runtime_kwargs
+            )
+        except Exception as exc:
+            comments.append(
+                'Failed to start new container \'{0}\': {1}'
+                .format(name, exc)
+            )
+            ret['comment'] = _format_comments(comments)
+            return ret
+
+        time.sleep(2)
```

Comment:
Good catch! That was in there for testing purposes when I was developing, and I forgot to remove it. I'll get it when I make the changes I mentioned [here](https://github.com/saltstack/salt/pull/22548#issuecomment-91996517).